lucene/highlighter/src/test/org/apache/lucene/search/uhighlight/TestUnifiedHighlighter.java

```diff
@@ -959,4 +961,84 @@ protected PassageFormatter getFormatter(String field) {
     ir.close();
   }
 
+  public void testBooleanWithSpanAndOverlappingTerms() throws IOException {
```

Comment:
Can you please simplify this one... it's ultimately testing preSpanQueryRewrite... maybe make it clear that it's testing that by naming the test method as-such?  Maybe this test should go into TestUnifiedHighlighterStrictPhrases as it's pertinent to that and not general stuff.